include/ginkgo/core/base/types.hpp

```diff
@@ -693,7 +758,7 @@ GKO_ATTRIBUTES constexpr bool operator!=(precision_reduction x,
 template <typename IndexType>
 inline constexpr GKO_ATTRIBUTES IndexType invalid_index()
 {
-    static_assert(std::is_signed<IndexType>::value,
+    static_assert(std::is_integral<IndexType>::value,
```

Comment:
Here you are changing the meaning of this function, so if you need an integral check, then you should probably have another function specific for that ?

Reply:
I'm reverting that, since it is not necessary anymore.